callbacks_test.go

```diff
@@ -465,6 +488,43 @@ func Test_ExecutorCallbacks(t *testing.T) {
 				})
 			})
 
+			Convey("notifies Sidecar to drain the service before stopping the watch looper", func() {
+				dummyWatchLooper := newMockLooper()
+				exec.watchLooper = &dummyWatchLooper
+				exec.watchLooper.Loop(func() error { return nil })
+
+				sidecarDrainChan = make(chan struct{})
+				quitChan := make(chan struct{})
+				go func() {
+					exec.KillTask(&dummyTaskID)
+					// Unblock this test
+					close(quitChan)
+				}()
+
+				// Wait a bit for exec.KillTask to do some work
+				time.Sleep(100 * time.Millisecond)
+
+				// Make sure the watch looper is not stopped before we block in the Sidecar draining handler
+				if dummyWatchLooper.isStopped {
```

Comment:
OK, can we not wait for an event on the `doneChan` for a normal `FreeLooper` instead of mocking this? Then you don't need the sleep, right? I feel like there is too much going on here with timing and stuff and want to get it to be simpler and not need the sleep.

Reply:
Actually, you're right! :) It always helps me to have somebody say that code looks too complex for no good reason. I got rid of that mock and managed to use a `FreeLooper` instead. Initially, I think I forgot that the looper needs to loop something in order to unblock `Wait()` after calling `Quit()` and I wasn't doing that. Please have another look.